vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/vaadin-grid-flow-selection-column.js

```diff
@@ -80,6 +80,10 @@ import { GridColumnElement } from '@vaadin/vaadin-grid/src/vaadin-grid-column.js
       if (this._grid) {
         this._grid.addEventListener('select', this._boundOnSelectEvent);
         this._grid.addEventListener('deselect', this._boundOnDeselectEvent);
+        // Prevent flickering when using the Lumo compact theme
+        if (this._grid.theme === "compact") {
```

Comment:
```suggestion
        if (this._grid.theme && this._grid.theme.split(' ').indexOf('compact') > -1) {
```

We need to check for multiple theme variants, as they can be used together.